frontend/src/views/Room.jsx

```diff
@@ -1,19 +1,168 @@
+  const tilesPerRow = {
+    xs: calculateTilesPerRow('xs'),
+    sm: calculateTilesPerRow('sm'),
+    md: calculateTilesPerRow('md'),
+  };
+  // initialize room
+  useEffect(() => {
+    const leaveRoom = async () => {
+      if (room) {
+        await userParticipant.unpublishAllTracks(); // also stops them
+        await room.leave();
+      }
+    };
+    const subscribeToRemoteStreams = async (r) => {
+      // subscribe ta all remote participants for testing purposes
+      const { remoteParticipants } = r;
+      const rps = Array.from(remoteParticipants.values());
+      await Promise.all(rps.map(async (rp) => {
+        await rp.subscribe();
+      }));
+      // console.log('subscribed to remote participant(s)');
+    };
+    const joinRoom = async () => {
+      const JWT = await roomJWTprovider(roomId, null, null, () => { setRoomNotFound(true); });
+      const newRoom = new WebRoom(JWT);
+      const newParticipant = await newRoom.join();
+
+      newRoom.on('ParticipantTrackSubscribed', (remoteParticipant, track) => {
+        const stream = new MediaStream();
+        stream.addTrack(track.mediaStreamTrack);
+        const streamObj = { stream, participantId: remoteParticipant.id };
+        setRemoteStreams([...remoteStreams, streamObj]);
+      });
+
+      // newRoom.on('ParticipantJoined', (p) => console.log('someone joined', p));
+      // newRoom.on('ParticipantLeft', (p) => console.log('someone left', p));
+
+      setRoom(newRoom);
+      const tracks = await newParticipant.publishTracks(
+        { constraints: { video: true, audio: true } },
+      );
+      const stream = new MediaStream();
+      tracks.forEach((track) => stream.addTrack(track.mediaStreamTrack));
+      setLocalStream(stream);
+      setUserParticipant(newParticipant);
+      subscribeToRemoteStreams(newRoom);
+    };
+    joinRoom();
+    return leaveRoom;
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+  const localStreamStyle = {
+    width: '25vw',
+    position: 'fixed',
+    bottom: 0,
+    right: 0,
+  };
   return (
-    <div>
-      This will eventually be a room with ID:
-      {' '}
-      {roomId}
-    </div>
+    <>
+      {
+        roomNotFound
+        && <Navigate to="/rooms/404" />
+      }
+      {
+        room ? (
+          <Box style={{ position: 'relative' }}>
+            <Grid sx={{ width: '65vw', height: '60vh' }} container spacing={2} columns={tilesPerRow} alignItems="center" justifyContent="center">
+              {
+                remoteStreams.map((stream) => (
+                  <Grid item xs={1} sm={1} md={1} key={stream.participantId}>
+                    <Box>
+                      <Video
+                        stream={stream.stream}
+                      />
+                    </Box>
+                  </Grid>
+                ))
+              }
+            </Grid>
+            <div style={localStreamStyle}>
+              <Video
+                stream={localStream}
+              />
```

Comment:
This will prevent the local user from hearing themselves:

```suggestion
              <Video
                 stream={localStream}
                 muted
               />
```